src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -0,0 +1,149 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+internal sealed class AsyncToSyncMethodTransformer : SyntaxTransformer {
+	public AsyncToSyncMethodTransformer(
+		SemanticModel model,
+		CancellationToken token
+	) : base( model, token ) { }

```

Comment:
I take the cancellation token in the constructor so that it won't have to be based down recursively (that's mostly the purpose of this class -- there is going to be _a lot_ of recursion here.)

See note below RE: tests and cancellation tokens.